Fix for security issue

maksimsinpuskins opened the pull request:
Problem: In code it sends request to https://via.placeholder.com/80x100?text=, this could lead to unwanted issues for example if this domain will die, change or it will be hacked, placeholder won't show or any other image will be shown in the app.

Fix: Add additional parameter in getThumbnail function to use it instead of request if it is not null.

Why: This fix will allow developers to prevent requests to API to create image placeholder by passing any widget to new additional parameter. Also, this placeholder could be created simply by using container with text widget in the center

Screenshot of example with opened .mp4 file with passed placeholder image as defaultImage
![Screenshot_2023-08-24-10-34-31-233_com example example_2](https://github.com/DutchCodingCompany/file_preview/assets/126455376/231bd702-5387-4693-be30-9841c99d958c)

Commit:
added optional Image parameter, changed example

Commit:
made parameter as Widget so developers can use any widget instead of only Image

NRijnberg commented:
Hi @maksimsinpuskins,

Thanks for your PR! Now you fixed the 'security issue' in the example app but not in the package itself. Can you add the asset in the package and only use the default asset when there is no image to render? This way we can get rid of the 'https://via.placeholder.com/80x100?text=$kExtension' completely. 

Also when checking out your pull request and running on iOS I encounter the following issue:

`CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
    [!] Unable to find a target named `RunnerTests` in project `Runner.xcodeproj`, did find `Runner`.

    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer/target_inspector.rb:108:in `compute_targets'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer/target_inspector.rb:40:in `compute_results'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer.rb:1197:in `block (3 levels) in inspect_targets_to_integrate'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer.rb:1195:in `each'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer.rb:1195:in `block (2 levels) in inspect_targets_to_integrate'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer.rb:1193:in `each'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer.rb:1193:in `block in inspect_targets_to_integrate'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/user_interface.rb:64:in `section'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer.rb:1188:in `inspect_targets_to_integrate'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer/analyzer.rb:107:in `analyze'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer.rb:422:in `analyze'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer.rb:244:in `block in resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/user_interface.rb:64:in `section'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer.rb:243:in `resolve_dependencies'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/installer.rb:162:in `install!'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/command/install.rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/lib/cocoapods/command.rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.15.2/bin/pod:55:in `<top (required)>'
    /usr/local/bin/pod:23:in `load'
    /usr/local/bin/pod:23:in `<main>'

Error output from CocoaPods:
↳
    Ignoring ffi-1.15.5 because its extensions are not built. Try: gem pristine ffi --version 1.15.5

    [!] Automatically assigning platform `iOS` with version `12.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.
`

When looking at the error 'Unable to find a target named `RunnerTests` in project `Runner.xcodeproj`, did find `Runner`'
a possible solution could be:
https://stackoverflow.com/questions/67405994/unable-to-find-a-target-named-runner-in-project-runner-xcodeproj-did-find-d

Commit:
Fix IOS build and add default image to package

maksimsinpuskins commented:
Hi @NRijnberg, added Podfile to git with removed RunnerTests target and added image to package itself, but I think that there still should be availability to choose custom default image as there could be cases when developers don't want users to see this exact image

NRijnberg commented on lib/file_previewer.dart:
```diff
@@ -64,14 +63,12 @@ class FilePreview {
 
   /// In case a file preview cannot be properly rendered, show a placeholder image with the extension in the center
-  static Image _defaultImage(String filePath) {
-    return Image.network(placeholderImageUrl(extension(filePath)));
```
Also remove placeholderImageUrl from the config.dart file. (Can delete entire file)

NRijnberg commented on lib/file_previewer.dart:
```diff
@@ -64,14 +63,12 @@ class FilePreview {
     }
   }
 
   /// In case a file preview cannot be properly rendered, show a placeholder image with the extension in the center
```
And no default image is provided

maksimsinpuskins replied:
so developers can't pass them in case they want to use custom one?

NRijnberg replied:
I meant to add it to the comment, so the comment would become:
In case a file preview cannot be properly rendered and no default image is provided, show a placeholder image with the........

maksimsinpuskins replied:
ah, sorry, misunderstood :sweat_smile:

NRijnberg replied:
No problem!

NRijnberg reviewed (commented):
Remove the assets from the pubspec in the example project:
`assets:
- assets/
`

 This now causes an error since this folder is deleted:
`Error: unable to find directory entry in pubspec.yaml: /path-to-package/file_preview/example/assets/`

NRijnberg commented:
> Hi @NRijnberg, added Podfile to git with removed RunnerTests target and added image to package itself, but I think that there still should be availability to choose custom default image as there could be cases when developers don't want users to see this exact image

I agree! I just meant to move the default asset into the package itself, the user providing their own widget is perfectly fine.

Commit:
removed config

Commit:
edit comment

NRijnberg reviewed (commented):
@maksimsinpuskins The only thing which remains to fix is the assets folder issue I mentioned above, after that I think it is ready

Commit:
fix assets is not found

NRijnberg commented:
@maksimsinpuskins  Thanks you for your PR! Merged it
